light_curves/lc_classifier.ipynb

```diff
@@ -0,0 +1,983 @@
+    "\n",
+    "## Introduction\n",
+    "This notebook takes output of a previous demo notebook which generates light curves from archival data, does data prep, and runs the light curves through multiple [`sktime`](https://www.sktime.net/en/stable/) classifiers.  The goal of the classifiers is to be able to differentiate changing look active galactic nucleii (CLAGN) from an SDSS quasar sample based on multiband light curves.  CLAGN are quite interested objects in that they appear to change state, but only a few hundred are currently known, and finding them is quite expensive requiring spectroscopic follow up.  Being able to identify CLAGN in existing large samples would allow us to identify a statisitcal sample from which we could better understand the physics of what is occuring in these systems.\n",
+    "\n",
+    "The challenges of this time-domain dataset are:\n",
+    "1. Multi-variate = There are multiple bands of observations per target (13+)\n",
+    "2. Unequal length = Each band has a light curve with different sampling than other bands\n",
+    "3. Missing data = Not each object has all observations in all bands\n",
+    "\n",
+    "We choose to use a Pandas multiindex dataframe to store and work with the data because it fulfills these requirements:\n",
+    "1. It can handle the above challenges of a dataset = multi-variate, unqueal length with missing data.\n",
+    "2. Multiple targets (multiple rows)\n",
+    "3. Pandas has some built in understanding of time units\n",
+    "4. Can be scaled up to big data numbers of rows (altough we don't push to out of memory structures in this use case)\n",
+    "5. Pandas is user friendly\n",
+    "\n",
+    "\n",
+    "## Input\n",
+    "Light curve parquet file of multiband light curves from the mulitband_lc.ipynb demo notebook.  The format of the light curves is a Pandas multiindex data frame\n",
+    "\n",
+    "A useful reference for what sktime expects as input to its ML algorithms: https://github.com/sktime/sktime/blob/main/examples/AA_datatypes_and_datasets.ipynb\n",
+    "\n",
+    "## Output\n",
+    "Trained classifiers as well as estimates of their accuracy and plots of confusion matrices\n",
+    "\n",
+    "## Imports\n",
+    "- `pandas` to work with light curve data structure\n",
+    "- `numpy` for numerical calculations\n",
+    "- `matplotlib` for plotting\n",
+    "- `sys` for paths\n",
+    "- `astropy` to work with coordinates/units and data structures\n",
+    "- `tqdm` for showing progress meter\n",
+    "- `sktime` ML algorithms specifically for time-domain data\n",
+    "- `sklearn` general use ML algorthims with easy to use interface\n",
+    "\n",
+    "## Authors\n",
+    "\n",
+    "## Acknowledgements"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "!pip install sktime\n",
+    "!pip install tqdm\n",
+    "!pip install googledrivedownloader"
```

Comment:
Does it make sense to add these to the requirements.txt? As the ML notebook requires packages such as acstools which is in the requirements anyways.

Reply:
yes, @troyraen @bsipocz @stargaser should we have a different requirements.txt file for each demo notebook in that light_curves directory?  Or all in one requirements.txt?

Reply:
I think it would make sense to have different requirements.txt for each use case, so we could, if needed use different version pins, but not necessarily one for each notebook (e.g. I expect the full use case to be able to run along with each other).
If if we run into some version requirement conflict, we could still separate it out to separate files per notebook.